From 69e23cd027d7b515c0526d9f9af0ee93da1b982e Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Mon, 15 Aug 2016 12:13:54 -0400
Subject: [PATCH 35/37] arm: dts: k2hk: add dts node for netcp-sa driver

This patch adds dts node for netcp-sa driver to support crypto inflow mode

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/keystone-k2hk-netcp.dtsi | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
index c45f229..8ee793d 100644
--- a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
@@ -314,7 +314,8 @@ netcp: netcp@2000000 {
 				<&dma_gbe 3>,	/* PA cluster 3 channel */
 				<&dma_gbe 4>,	/* PA cluster 4 channel */
 				<&dma_gbe 5>,	/* PA cluster 5 channel */
-				<&dma_gbe 31>;	/* PA rx flow for cmd rsp */
+				<&dma_gbe 31>,	/* PA rx flow for cmd rsp */
+				<&dma_gbe 6>;	/* SA tx channel */
 	ti,navigator-dma-names = "netrx0",
 				 "netrx1",
 				 "nettx",
@@ -326,7 +327,8 @@ netcp: netcp@2000000 {
 				 "pa-cl3",
 				 "pa-cl4",
 				 "pa-cl5",
-				 "pa-cmd-rsp";
+				 "pa-cmd-rsp",
+				 "sa-tx0";
 
 	netcp-devices {
 		#address-cells = <1>;
@@ -462,6 +464,26 @@ netcp: netcp@2000000 {
 				};
 			};
 		};
+
+		sa@c0000 {
+			label = "netcp-sa";
+			multi-interface;
+			/* Optional: Specify NATT port for fragmentation detection */
+			natt-port = <0>;
+			tx-submit-queue = <646>;
+
+			interfaces {
+				sa0: interface-0 {
+					slave-port	= <0>;
+					ti,tx-channel = "sa-tx0";
+				};
+
+				sa1: interface-1 {
+					slave-port	= <1>;
+					ti,tx-channel = "sa-tx0";
+				};
+			};
+		};
 	};
 
 	netcp-interfaces {
@@ -486,6 +508,7 @@ netcp: netcp@2000000 {
 			netcp-gbe = <&gbe0>;
 			netcp-pa = <&pa0>;
 			netcp-qos = <&qos0>;
+			netcp-sa = <&sa0>;
 		};
 		interface-1 {
 			rx-channel = "netrx1";
@@ -506,6 +529,7 @@ netcp: netcp@2000000 {
 			netcp-gbe = <&gbe1>;
 			netcp-pa = <&pa1>;
 			netcp-qos = <&qos1>;
+			netcp-sa = <&sa1>;
 		};
 	};
 };
-- 
1.9.1

